sdk/src/main/java/io/radar/sdk/RadarInAppMessageManager.kt

```diff
@@ -0,0 +1,145 @@
+package io.radar.sdk
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.net.toUri
+import io.radar.sdk.Radar.RadarLogConversionCallback
+import io.radar.sdk.model.RadarEvent
+import io.radar.sdk.model.RadarInAppMessage
+import org.json.JSONObject
+
+class RadarInAppMessageManager(private val activity: Activity, private val context: Context) {
+    private var currentView: View? = null
+    private var inAppMessageReceiver: RadarInAppMessageReceiver? = null
+
+    // Time tracking properties
+    private var modalShowTime: Long = 0L
+    private var modalDismissTime: Long = 0L
+
+    private fun showModal(payload: RadarInAppMessage) {
+
+        if (activity == null) {
+            Radar.logger.e("Activity is null, cannot show in-app message")
+            return
+        }
+
+        // Record the time when modal is shown
+        modalShowTime = System.currentTimeMillis()
+        
+        val metadata = makeConversionMetadata(payload)
+        Radar.sendLogConversionRequest("in_app_message_displayed", metadata, callback = object : RadarLogConversionCallback {
+            override fun onComplete(status: Radar.RadarStatus, event: RadarEvent?) {
+                Radar.logger.i("Conversion name = ${event?.conversionName}: status = $status; event = $event")
+            }
+        })
```

Comment:
it's possible for the view to get created but now shown (in a race condition). So this conversion log should be placed in onViewReady after setting currentView